src/components/shop-detail.js

```diff
@@ -0,0 +1,230 @@
+import { LitElement, html } from '../../node_modules/@polymer/lit-element/lit-element.js';
+import { unsafeHTML } from '../../node_modules/lit-html/lib/unsafe-html.js';
+import { shopButtonStyle } from './shop-button-style.js';
+import { shopCommonStyle } from './shop-common-style.js';
+import { shopSelectStyle } from './shop-select-style.js';
+import './shop-image.js';
+
+import { store } from '../store.js';
+import { connect } from '../../node_modules/redux-helpers/connect-mixin.js';
+import { splitPathSelector } from '../reducers/location.js';
+import { currentCategorySelector, currentItemSelector } from '../reducers/categories.js';
+import { addCartEntry } from '../actions/cart.js';
+
+class ShopDetail extends connect(store)(LitElement) {
+  render({ failure, item }) {
+    return html`
+    <style>
+      ${shopButtonStyle}
+      ${shopCommonStyle}
+      ${shopSelectStyle}
+
+      :host {
+        display: block;
+      }
+
+      #content {
+        display: flex;
+        flex-direction: row;
+        justify-content: center;
+      }
+
+      shop-image {
+        position: relative;
+        margin: 64px 32px;
+        width: 50%;
+        max-width: 600px;
+      }
+
+      shop-image::before {
+        content: "";
+        display: block;
+        padding-top: 100%;
+      }
+
+      .detail {
+        margin: 64px 32px;
+        width: 50%;
+        max-width: 400px;
+        transition: opacity 0.4s;
+        opacity: 0;
+      }
+
+      .detail[has-content] {
+        opacity: 1;
+      }
+
+      h1 {
+        font-size: 24px;
+        font-weight: 500;
+        line-height: 28px;
+        margin: 0;
+      }
+
+      .price {
+        margin: 16px 0 40px;
+        font-size: 16px;
+        color: var(--app-secondary-color);
+      }
+
+      .description {
+        margin: 32px 0;
+      }
+
+      .description > h2 {
+        margin: 16px 0;
+        font-size: 13px;
+      }
+
+      .description > p {
+        margin: 0;
+        color: var(--app-secondary-color);
+      }
+
+      .pickers {
+        display: flex;
+        flex-direction: column;
+        border-top: 1px solid #ccc;
+      }
+
+      shop-select > select {
+        font-size: 16px;
+        padding: 16px 24px 16px 70px;
+      }
+
+      @media (max-width: 767px) {
+
+        #content {
+          flex-direction: column;
+          align-items: center;
+        }
+
+        shop-image {
+          margin: 0;
+          width: 80%;
+        }
+
+        .detail {
+          box-sizing: border-box;
+          margin: 32px 0;
+          padding: 0 24px;
+          width: 100%;
+          max-width: 600px;
+        }
+
+        h1 {
+          font-size: 20px;
+          line-height: 24px;
+        }
+
+        .price {
+          font-size: inherit;
+          margin: 12px 0 32px;
+        }
+
+      }
+
+    </style>
+
+    ${ !failure && item ? html`
+      <div id="content">
```

Comment:
This will restamp the template if `item` changes from `{..}` -> `null` -> `{..}`. So this is not very efficient. Lit-html has plan to support `when` directive which is what you want. But before we have that a better way would be to just show/hide `div.#content` and `shop-network-warning`.